app/commands/user/set_discourse_groups.rb

```diff
@@ -0,0 +1,39 @@
+class User::SetDiscourseGroups
+  include Mandate
+
+  initialize_with :user
+
+  def call
+    set_trust_level!
+    set_pm_enabled!
+  rescue DiscourseApi::NotFoundError
+    # If the external user can't be found, then the
+    # oauth didn't complete so there's nothing to do.
+  end
+
+  private
+  def set_trust_level!
+    return unless user.reputation >= MIN_REP_FOR_TRUST_LEVEL
```

Comment:
`if` is arguably slightly easier to read than `unless`:

```suggestion
    return if user.reputation < MIN_REP_FOR_TRUST_LEVEL
```